packages/mobx/src/api/observe.ts

```diff
@@ -25,17 +25,17 @@ export function observe<T>(
     fireImmediately?: boolean
 ): Lambda
 export function observe<V>(
-    observableMap: ObservableSet<V>,
+    observableMap: ObservableSet<V> | Set<V>,
```

Comment:
Is the observable variant `ObservableSet/Map` even good for anything here? It implements `Set`...

Reply:
that seems problematic indeed, as `observableMap` would effectively become the least common denominator of both. Why was this needed?

Reply:
> Why was this needed?

In case you're asking about the need for `Set`/`Map`, there is an overload for object, which doesn't require `ObservableObject` but any `Object`:
```javascript
export function observe(
    object: Object,
    listener: (change: IObjectDidChange) => void,
    fireImmediately?: boolean
): Lambda
```
So Map/Set requires explicit cast to `ObservableMap/Set` otherwise it matches object overload. That's inconsistent and error prone.

Reply:
Check!

Reply:
@urugator Thanks, I've removed ObservableMap and ObservableSet from the typings. For some reason tests with ObservableMap pass but tests with ObservableSet don't:

![image](https://user-images.githubusercontent.com/22447849/141111386-1723240b-0d82-4e7c-be81-a2a6048afcc6.png)

This issue on CI: https://app.circleci.com/pipelines/github/mobxjs/mobx/1788/workflows/4a10412e-e197-4eea-bde8-3221d9b39c31/jobs/8295

I've double checked that both ObservableMap/ObservableSet implement ES6 versions. Both of them have same overload:

![image](https://user-images.githubusercontent.com/22447849/141111640-198856e6-40f9-4d49-b695-302755b2414b.png)

So the reason of issue is not clear. Workarounds:
- Return ObservableSet/ObservableMap back
- change unit test to match `ISetDidChange<any>`

Reply:
There are TS setups where Set/Map is not defined (based on the selected standard libs), so that union might not be that bad after all

Reply:
If that's the case I think a explanatory (or link) comment would be nice.